# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/OffscreenCanvas.h
# Commit: bbfe23c61add
# Full Hash: bbfe23c61add07377daf80696fd6996c0463f5b6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 8fcfac450ecc -r bbfe23c61add dom/canvas/OffscreenCanvas.h
--- a/dom/canvas/OffscreenCanvas.h	Mon Jun 08 20:13:01 2020 +0000
+++ b/dom/canvas/OffscreenCanvas.h	Mon Jun 08 20:34:15 2020 +0000
@@ -21,8 +21,8 @@
 class ErrorResult;
 
 namespace layers {
-class AsyncCanvasRenderer;
 class CanvasClient;
+class CanvasRenderer;
 class ImageContainer;
 }  // namespace layers
 
@@ -35,13 +35,13 @@
 // Canvas to worker thread directly. Thus, we create this helper class and
 // store necessary data in it then pass it to worker thread.
 struct OffscreenCanvasCloneData final {
-  OffscreenCanvasCloneData(layers::AsyncCanvasRenderer* aRenderer,
-                           uint32_t aWidth, uint32_t aHeight,
+  OffscreenCanvasCloneData(layers::CanvasRenderer* aRenderer, uint32_t aWidth,
+                           uint32_t aHeight,
                            layers::LayersBackend aCompositorBackend,
                            bool aNeutered, bool aIsWriteOnly);
   ~OffscreenCanvasCloneData();
 
-  RefPtr<layers::AsyncCanvasRenderer> mRenderer;
+  RefPtr<layers::CanvasRenderer> mRenderer;
   uint32_t mWidth;
   uint32_t mHeight;
   layers::LayersBackend mCompositorBackendType;
@@ -58,7 +58,7 @@
 
   OffscreenCanvas(nsIGlobalObject* aGlobal, uint32_t aWidth, uint32_t aHeight,
                   layers::LayersBackend aCompositorBackend,
-                  layers::AsyncCanvasRenderer* aRenderer);
+                  layers::CanvasRenderer* aRenderer);
 
   nsCOMPtr<nsIGlobalObject> GetParentObject() const { return GetOwnerGlobal(); }
 
@@ -170,7 +170,7 @@
   layers::LayersBackend mCompositorBackendType;
 
   RefPtr<layers::CanvasClient> mCanvasClient;
-  RefPtr<layers::AsyncCanvasRenderer> mCanvasRenderer;
+  RefPtr<layers::CanvasRenderer> mCanvasRenderer;
 };
 
 }  // namespace dom